make-testsuite/Makefile

```diff
@@ -0,0 +1,95 @@
+MKDIR ?= mkdir
+
+OBJ = obj
+
+WOLFSSH ?= wolfssh
+SSHDIR = $(WOLFSSH)/src
+SSHINC = $(WOLFSSH)
+OBJSSH = $(OBJ)/$(WOLFSSH)
+
+WOLFSSL ?= wolfssl
+CRYPTDIR = $(WOLFSSL)/wolfcrypt/src
+CRYPTINC = $(WOLFSSL)
+OBJCRYPT = $(OBJ)/$(WOLFSSL)
+
+CPPFLAGS ?= -I. -I$(SSHINC) -I$(CRYPTINC) -DWOLFSSL_USER_SETTINGS
+ifeq ($(BUILD),debug)
+    DEBUG ?= -O0 -g -DDEBUG_WOLFSSH
+endif
+CFLAGS := $(DEBUG) $(CFLAGS)
+
+LDFLAGS ?= -lm -pthread
+
+.PHONY: clean all
+
+all: $(OBJ) libwolfssh.a testsuite keys/server-key-rsa.der
+
+testsuite: $(OBJ)/testsuite.o $(OBJ)/echoserver.o $(OBJ)/client.o libwolfssh.a
+	$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS)
+
+libwolfssh.a: $(OBJSSH)/agent.o $(OBJSSH)/keygen.o $(OBJSSH)/port.o \
+  $(OBJSSH)/wolfsftp.o $(OBJSSH)/internal.o $(OBJSSH)/log.o $(OBJSSH)/ssh.o \
+  $(OBJSSH)/wolfterm.o $(OBJSSH)/io.o $(OBJSSH)/wolfscp.o \
+  $(OBJCRYPT)/aes.o $(OBJCRYPT)/dh.o $(OBJCRYPT)/integer.o $(OBJCRYPT)/tfm.o \
+  $(OBJCRYPT)/sha.o $(OBJCRYPT)/sha256.o $(OBJCRYPT)/sha512.o \
+  $(OBJCRYPT)/hash.o $(OBJCRYPT)/ecc.o $(OBJCRYPT)/rsa.o $(OBJCRYPT)/memory.o \
+  $(OBJCRYPT)/random.o $(OBJCRYPT)/hmac.o $(OBJCRYPT)/wolfmath.o \
+  $(OBJCRYPT)/asn.o $(OBJCRYPT)/coding.o $(OBJCRYPT)/signature.o \
+  $(OBJCRYPT)/wc_port.o
+	$(AR) $(ARFLAGS) $@ $^
+
+$(OBJSSH)/%.o: $(SSHDIR)/%.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $<
+
+$(OBJSSH)/agent.o: $(SSHDIR)/agent.c
+$(OBJSSH)/keygen.o: $(SSHDIR)/keygen.c
+$(OBJSSH)/port.o: $(SSHDIR)/port.c
+$(OBJSSH)/wolfsftp.o: $(SSHDIR)/wolfsftp.c
+$(OBJSSH)/internal.o: $(SSHDIR)/internal.c
+$(OBJSSH)/log.o: $(SSHDIR)/log.c
+$(OBJSSH)/ssh.o: $(SSHDIR)/ssh.c
+$(OBJSSH)/wolfterm.o: $(SSHDIR)/wolfterm.c
+$(OBJSSH)/io.o: $(SSHDIR)/io.c
+$(OBJSSH)/wolfscp.o: $(SSHDIR)/wolfscp.c
+
+$(OBJCRYPT)/%.o: $(CRYPTDIR)/%.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c -o $@ $<
+
+$(OBJCRYPT)/aes.o: $(CRYPTDIR)/aes.c
+$(OBJCRYPT)/dh.o: $(CRYPTDIR)/dh.c
+$(OBJCRYPT)/tfm.o: $(CRYPTDIR)/tfm.c
```

Comment:
Should you also include sp_int.c and integer.c to accommodate math lib changes to the user_settings.h?

Reply:
I had to add sp_c64 and sp_c32, just in case. There is some direction on setting the math library in the user-settings.